### DISCUSSION TITLE
Detect duplicates from messaging apps in albums

### DISCUSSION THREAD

**Discussion by jagjordi** [immich-app/immich#2082] [General]
If i share a image with my partner through WhatsApp and we both sync all our phone library to immich then those shared pictures are duplicated. When you try to create an album you want that the original (not the compressed Whatsapp one) is the one added to the album.

It would be very nice to have a feature that would remove the duplicates using [image hash](https://github.com/JohannesBuchner/imagehash) or a similar mechanism.  

Example of usage:
1. Both users add pictures to an album including duplicates from Whatsapp or similar.
2. Inside the album there is a button to "remove low quality duplicates" or similar
3. When that option is pressed all the images are checked with their hash for duplicates and if a duplicate is found the lower quality asset is removed from the album

**Comment by bo0tzz** [comment]
Are you sharing an account?

**Comment by jagjordi** [comment]
You mean in immich?

**Reply by bo0tzz** [reply]
Yes

**Comment by jagjordi** [comment]
We use different accounts in immich

Best regards,

Jordi Altayó
________________________________

**Comment by feerlessleadr** [comment]
I'm facing a similar issue (but not exactly the same, so if needed I can open a new thread). My wife and I share an Immich account, and when she takes a picture and sends me that picture in a text message, both the picture she took and the picture in the text message I receive on my phone are uploaded to immich, causing a duplicate. 

I'm assuming this is because the file names are different, hence they are not being detected as duplicate, but I'm not sure how the duplicate process works to be sure.

**Reply by alextran1502** [reply]
@feerlessleadr I think this is a different issue. The picture you download might be compressed by Apple. Can you go to Immich's app settings and turn on advanced troubleshooting mode and go to a set of duplicate image, open the info view and share with us the screenshot from the two assets?

**Reply by feerlessleadr** [reply]
Thanks - I think you're right (although I'm on Android, but I believe the same scenario applies). The first picture is the sms version (that I recieved), the second is the original that my wife took and was backed up to immich directly. 
![sms](https://user-images.githubusercontent.com/288895/233685446-782f74eb-0b06-471e-9fcd-48cc47252084.jpg)
![original](https://user-images.githubusercontent.com/288895/233685448-108de6ca-2568-4375-ac4f-24306de98c33.jpg)
@alextran1502

**Reply by alextran1502** [reply]
@feerlessleadr from the width and height of the images you can see the sms version is compressed. So the hash is different. Is there any folder on Android that store only assets received from sms? You can exclude that album not be uploaded

**Reply by feerlessleadr** [reply]
Thanks - this is definitely an issue on my end then. I had explicitly added this in to immich to be backed up since I'd like to backup the pictures I receive (at least from people other than my wife at this point). 

I'll continue to play with it - thanks!

**Reply by x42x64** [reply]
I'm running into the same thing with WhatsApp (it's compressed through it as well). Even worse - as the images in WhatsApp have different Metadata, the same image appears even at different points in time. I want to backup/store those photos as well. I'm not sure how to deal with it, maybe someone else has an idea...

**Reply by konradr25** [reply]
It would be nice to have feature to remove duplicates.

**Comment by TheOriginal92** [comment]
Second this. Duplicate detection would be very very useful!
